backend/compact-connect/lambdas/python/common/cc_common/data_model/transaction_client.py

```diff
@@ -128,3 +119,67 @@ def _query_transactions_for_month(
                 break
 
         return all_matching_transactions
+
+    def _set_privilege_id_in_line_item(self, line_items: list[dict], item_id: str, privilege_id: str):
+        for line_item in line_items:
+            if line_item.get('itemId').lower() == item_id.lower():
+                line_item['privilegeId'] = privilege_id
+
+    def add_privilege_ids_to_transactions(self, compact: str, transactions: list[dict]) -> list[dict]:
+        """
+        Add privilege IDs to transaction line items based on the jurisdiction they were purchased for.
+
+        :param compact: The compact name
+        :param transactions: List of transaction records to process
+        :return: Modified list of transactions with privilege IDs added to line items
+        """
+        for transaction in transactions:
+            line_items = transaction['lineItems']
+            # Extract jurisdictions from line items with format priv:{compact}-{jurisdiction}
+            jurisdictions_to_process = set()
+            for line_item in line_items:
+                item_id = line_item['itemId']
+                if item_id.startswith('priv:'):
+                    jurisdiction = item_id.split('-')[1].lower()
+                    jurisdictions_to_process.add(jurisdiction)
+
+            # Query for privilege records using the GSI
+            gsi_pk = f'COMPACT#{compact}#TX#{transaction["transactionId"]}#'
+            response = self.config.provider_table.query(
+                IndexName=self.config.compact_transaction_id_gsi_name,
+                KeyConditionExpression=Key('compactTransactionIdGSIPK').eq(gsi_pk),
+            )
+
+            # Process each privilege record
+            for jurisdiction in jurisdictions_to_process:
+                item_id = f'priv:{compact}-{jurisdiction}'
+                # find the first privilege record for the jurisdiction
+                matching_privilege = next(
```

Comment:
Let's add `providerId` to the GSI projected attributes, so we can include a sanity check here to verify that the privilege we matched is actually associated with the individual we expect.